File: wifi_manager.py
import time

class WifiManager:
    def __init__(self, ssid, password):
        self.ssid = ssid
        self.password = password
        self.wlan = network.WLAN(network.STA_IF)

    def connect(self):
        if not self.wlan.isconnected():
            print('Connecting to network...', end='')
            self.wlan.active(True)
            self.wlan.connect(self.ssid, self.password)

            max_wait = 10
            while max_wait > 0:
                if self.wlan.status() < 0 or self.wlan.status() >= 3:
                    break
                max_wait -= 1
                print('.', end='')
                time.sleep(1)
            
            print()

        if self.wlan.isconnected():
            print('Network config:', self.wlan.ifconfig())
            return True 

File: wifi_manager.cpp
#include "wifi_manager.h"
#include <Arduino.h>
#include <WiFi.h>

WifiManager::WifiManager(const char* ssid, const char* password)
: _ssid(ssid), _password(password) {
}

void WifiManager::connect() {
    Serial.print("Connecting to network: ");
    Serial.println(_ssid);

    WiFi.begin(_ssid, _password);

    int max_wait = 20; // Час очікування підключення в секундах (20 * 500ms = 10s)
    while (WiFi.status() != WL_CONNECTED && max_wait > 0) {
        delay(500);
        Serial.print(".");
        max_wait--;
    }

    Serial.println();

    if (WiFi.status() == WL_CONNECTED) {
        Serial.println("WiFi connected!");
        Serial.print("IP Address: ");
        Serial.println(WiFi.localIP());
    } else {
        Serial.println("WiFi connection failed!");
    }
} 